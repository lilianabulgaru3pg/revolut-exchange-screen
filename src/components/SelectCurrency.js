import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import MenuItem from '@material-ui/core/MenuItem';
import { Grid } from '@material-ui/core';

const Select = withStyles({
  root: {
    width: '6rem',
    '& .MuiInput-underline': {
      '&:hover:before': {
        borderBottom: 'none',
      },
      '&:before': {
        borderBottom: 'none',
      },
    },
  },
})(TextField);

const SelectCurrency = ({
  id,
  currency,
  currencies,
  onCurrencyChange,
}) => (
  <Grid container>
    <Grid item sm={12}>
      <Select
        id={id}
        select
        value={currency}
        onChange={e => onCurrencyChange(e.target.value)}
      >
        {Object.keys(currencies).map(key => (
          <MenuItem key={key} value={currencies[key]}>
            {currencies[key]}
          </MenuItem>
        ))}
      </Select>
    </Grid>
  </Grid>
);

export default SelectCurrency;
