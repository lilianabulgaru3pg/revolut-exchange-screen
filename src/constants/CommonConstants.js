const CURRENCIES = {
  EUR: 'EUR',
  USD: 'USD',
  GBP: 'GBP',
};

const SYMBOL = {
  EUR: '€',
  USD: '$',
  GBP: '£',
};

const EB_BASE_URL = 'https://api.exchangeratesapi.io';

const ENABLE_EXCHANGE_BTN = 'ENABLE_EXCHANGE_BTN';
const DISABLE_EXCHANGE_BTN = 'DISABLE_EXCHANGE_BTN';

const DEFAULT_TO_CURRENCY = 'EUR';
const USER_DEFAULT_BASE_CURRENCY = 'GBP';

const SELECT_BASE_CURRENCY = 'SELECT_BASE_CURRENCY';
const SELECT_TO_CURRENCY = 'SELECT_TO_CURRENCY';

const availableBalace = 500;
const exchangeTitle = 'Exchange';

const inputFromType = 'from';
const inputToType = 'to';

const FETCH_INTERVAL = 10000;

export {
  SYMBOL,
  CURRENCIES,
  FETCH_INTERVAL,
  DEFAULT_TO_CURRENCY,
  USER_DEFAULT_BASE_CURRENCY,
  SELECT_BASE_CURRENCY,
  SELECT_TO_CURRENCY,
  availableBalace,
  exchangeTitle,
  ENABLE_EXCHANGE_BTN,
  DISABLE_EXCHANGE_BTN,
  EB_BASE_URL,
  inputFromType,
  inputToType,
};
