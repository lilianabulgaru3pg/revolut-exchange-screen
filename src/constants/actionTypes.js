export const SELECT_BASE_CURRENCY = 'SELECT_BASE_CURRENCY';
export const SELECT_TO_CURRENCY = 'SELECT_TO_CURRENCY';

export const SELECT_BASE_SYMBOL = 'SELECT_BASE_SYMBOL';
export const SELECT_TO_SYMBOL = 'SELECT_TO_SYMBOL';

export const DISABLE_EXCHANGE_BTN = 'DISABLE_EXCHANGE_BTN';
export const ENABLE_EXCHANGE_BTN = 'ENABLE_EXCHANGE_BTN';

export const EXCHANGE = 'EXCHANGE';
export const EXCHANGE_BALANCE = 'EXCHANGE_BALANCE';

export const RATES_FETCH = 'RATES_FETCH';
export const RATES_FETCH_ERROR = 'RATES_FETCH_ERROR';

export const RATES_ADD = 'RATES_ADD';

export const INPUT_FROM = 'INPUT_FROM';
export const INPUT_TO = 'INPUT_TO';
export const DISPLAY_INPUT_FROM = 'CHANGE_INPUT_FROM';
export const DISPLAY_INPUT_TO = 'CHANGE_INPUT_TO';
export const RESET_INPUTS = 'RESET_INPUTS';
