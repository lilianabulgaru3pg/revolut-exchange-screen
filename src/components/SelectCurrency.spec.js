import React from 'react';
import renderer from 'react-test-renderer';
import SelectCurrency from './SelectCurrency';
import { CURRENCIES } from '../constants/CommonConstants';

describe('SelectCurrency Component', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <SelectCurrency
        id="select-base-currency"
        currency={CURRENCIES.EUR}
        currencies={CURRENCIES}
        onCurrencyChange={jest.fn()}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
