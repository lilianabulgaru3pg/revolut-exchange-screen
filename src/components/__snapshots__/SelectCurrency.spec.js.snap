// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectCurrency Component snapshot renders 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
  >
    <div
      className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-104"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby=" select-base-currency"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="select-base-currency"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          EUR
        </div>
        <input
          autoFocus={false}
          onAnimationStart={[Function]}
          type="hidden"
          value="EUR"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
